github-runner-manager/src/github_runner_manager/reactive/consumer.py

```diff
@@ -125,6 +127,14 @@ def consume(  # noqa: C901
                 if msg.payload == END_PROCESSING_PAYLOAD:
                     msg.ack()
                     break
+
+                msg.headers[PROCESS_COUNT_HEADER_NAME] = (
+                    msg.headers.get(PROCESS_COUNT_HEADER_NAME, 0) + 1
+                )
```

Comment:
Out of curiosity, does setting the message headers get reflected back in the queue? If not, should we keep a local variable for it?

The reason is because this snippet of code seems to suggest that the message will be modified - which is going to be shared over other workers that receive the message, but I don't quite see that happening further down the code.

Reply:
As I see it - I don't quite see a case where we reject the message with a requeue. Would this be intended? Or would this be a preparation for the future case where we would requeue?

Reply:
The rejection is done at two places:
- sigterm signal handler.
- In the spawning runner function at the of this function. It would wait for the reactive job to be picked up, if not it gets requeued.